src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -77,37 +77,49 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $query->condition($bundle_key, $bundles, 'IN');
     }
 
-    $user_groups = $this->getUserGroups();
-    if (!$user_groups) {
+    $user = User::load(\Drupal::currentUser()->id());
+
+    if (!$user) {
+      // @todo: What to do with anonymous user?
       return $query;
     }
 
+    if ($user->hasPermission('administer group')) {
+      // User can see all the groups.
+      return $query;
+    }
+
+    // User is a non-site wide group admin.
     $identifier_key = $definition->getKey('id');
 
+    /** @var \Drupal\og\OgAccessInterface $og_access */
+    $og_access = \Drupal::service('og.access');
```

Comment:
Ohh sorry. I thought this is the field constraints when this is the field plugin. I'll fix first the merge and import the regression tests from D7. The injection I'll handle in the end.